packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -639,10 +642,33 @@ export function TransferPanel() {
         isBatchTransferSupported &&
         Number(amount2) > 0
 
+      let nativeCurrencyToApprove = amountBigNumber
+
+      // Eth transfers to a custom destination use retryables
+      // In the case of native currency we need to also approve native currency used for gas
+      if (!selectedToken && isCustomDestinationTransfer) {
+        const retryableGasEstimates =
+          await bridgeTransferStarter.transferEstimateGas({
+            amount: amountBigNumber,
+            signer
+          })
+
+        const parentGasPrice = await parentChainProvider.getGasPrice()
+
+        const parentRetryableGas =
+          retryableGasEstimates?.estimatedParentChainGas.mul(parentGasPrice)
+
+        if (parentRetryableGas) {
+          nativeCurrencyToApprove = nativeCurrencyToApprove.add(
+            percentIncrease(parentRetryableGas, BigNumber.from(30))
+          )
+        }
+      }
+
       const isNativeCurrencyApprovalRequired =
         await bridgeTransferStarter.requiresNativeCurrencyApproval({
           signer,
-          amount: amountBigNumber,
+          amount: nativeCurrencyToApprove,
```

Comment:
this is a confusing name because this `amount` is potentially the selected token's amount

to handle the custom destination address native currency transfer case's approval, I think it's better to add `options` to the `EthDepositStarter`'s `requiresNativeCurrencyApproval` and have the relevant code there

Reply:
updated to use `approvalAmountIncrease`